src/main/java/org/folio/uk/service/UserService.java

```diff
@@ -63,15 +66,24 @@ public class UserService {
   private final UserPermissionsClient userPermissionsClient;
   private final FolioExecutionContext folioExecutionContext;
   private final RolesKeycloakConfigurationProperties rolesKeycloakConfiguration;
+  private final KeycloakFederatedAuthProperties keycloakFederatedAuthProperties;
 
   public User createUser(User user, boolean keycloakOnly) {
     return createUser(user, null, keycloakOnly);
   }
 
   public User createUser(User user, String password, boolean keycloakOnly) {
-
     return createUserPrivate(user, keycloakOnly, this::createUserInUserServiceSafe,
-      createdUser -> keycloakService.createUser(createdUser, password));
+      createdUser -> {
+        if (Objects.isNull(user.getId())) {
```

Comment:
this validation checks user object, maybe it was supposed to check createdUser user?
also please check if we do not have good reason to introduce this validation lets not introduce as it worked before